### PR TITLE
Add Shopping Cart to Docs (Examples.md)

### PR DESCRIPTION
**Pull request by justingreenberg**
@gaearon I referenced some patterns you described in your PR, can you take a quick look?

redux-shopping-cart-docs

With respect to the Examples in general, maybe we could replace "It covers:" with something like "Patterns:" and try to include links to relevant code in bullet points. Since many people use these examples as learning tools and references, the ability to scan and jump to patterns would be helpful.

### CODE REVIEW

**Comment by gaearon** [comment]
I like your suggestion about links.
But what do we link to? Master? Line offsets may change. Particular commit? It will get outdated.

**Comment by justingreenberg** [comment]
Unfortunately it's hard to imagine an automated, bulletproof method for maintaining these inter-repo reference links—other than diligence on the part of contributors and community. Ideally, any patterns referenced in examples would be idiomatic redux with a long shelf life.

Either way, the problem of maintaining inter-repo code references an interesting one. Give me a little bit to think more about it and get back to you... In the meantime, do you want to leave the reference links in this PR? Obviously I will maintain them for this example :wink:

**Comment by hedgerh** [comment]
@justingreenberg What would you do if a concept/pattern wasn't just a single line/block of code?  What would you link to if it spanned multiple files?

**Comment by hedgerh** [comment]
Kicking out ideas here, but one option, if that is an issue, would be to link to a gist or something that showed the concept, then the user could see for themselves where its being applied in the example, since the examples don't take very long to look through.

That way, we wouldn't need to maintain references to specific lines of code, and the code wouldn't necessarily need to match 1-to-1 with the example.

**Comment by hedgerh** [comment]
Also, could you add a link to the shopping cart example into README.md in this PR?

**Comment by gaearon** [comment]
Let's remove the links to the lines of code for now.
I think they'll be a pain to maintain, and we also want to be consistent with other examples.

File: README.md [hunk 1]
```diff
@@ -155,12 +156,12 @@ Special thanks to [Jamie Paton](http://jdpaton.github.io) for handing over the `
...
 ### Change Log
 
-This project adheres to [Semantic Versioning](http://semver.org/).  
+This project adheres to [Semantic Versioning](http://semver.org/).
 Every release, along with the migration instructions, is documented on the Github [Releases](https://github.com/rackt/redux/releases) page.
 
 ### Patrons
 
-The work on Redux was [funded by the community](https://www.patreon.com/reactdx).  
+The work on Redux was [funded by the community](https://www.patreon.com/reactdx).
```
Comment (gaearon) [inline README.md]: Please don't remove trailing spaces in README.
Whitespace is significant in Markdown.
These two spaces at the end of the line tell it to break the line.

Comment (justingreenberg) [inline README.md]: Sorry! Sublime trimmed the trailing whitespace, I didn't notice on commit. Rebasing now.

**Comment by justingreenberg** [comment]
@gaearon removed links and fixed whitespace issue @hedgerh added shopping cart to README

**Comment by hedgerh** [comment]
LGTM :+1:

**Comment by gaearon** [comment]
Thank you.

**Pull request merged**